src/collections/utils.zig

```diff
@@ -5,12 +5,16 @@ const std = @import("std");
 /// Since there's nothing we can do to recover from such an issue,
 /// it's best to always exit the process with a nice message than to
 /// propagate unrecoverable errors all over the compiler.
-pub fn exitOnOom() noreturn {
-    const oom_message =
-        \\I ran out of memory! I can't do anything to recover, so I'm exiting.
-        \\Try reducing memory usage on your machine and then running again.
-    ;
+pub fn exitOnOom(err: std.mem.Allocator.Error) noreturn {
+    switch (err) {
```

Comment:
What do we expect to happen if we pass in an error that isn't OutOfMemory? I think I missed some context. I noticed your comment in zulip

> Use Andrew Kelley's suggestion for catching OOM errors by capturing the Allocator.Error instead of discarding it, this should prevent us from accidentally over/under-catching errors

Reply:
I've noticed we have called it differently in some places, so just trying to understand more about that. 

````zig
switch (err) {
    error.OutOfMemory => exitOnOom(error.OutOfMemory),
    else => return err,
}

catch |err| exitOnOom(err)
```

Reply:
That would lead to a compile error, like if you passed in `error.FileNotFound` it would fail to build. You can manually pass `error.OutOfMemory` if you want to trigger this, but we don't really need that yet.

Reply:
Neat